python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_text_embedding.py

```diff
@@ -0,0 +1,106 @@
+        endpoint: str | None = None,
+        service_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+        client: EmbeddingsClientAsync | None = None,
+        model_info: ModelInfo | None = None,
+    ) -> None:
+        """Initialize the Azure AI Inference Text Embedding service.
+
+        Args:
+            api_key: The API key for the Azure AI Inference service deployment.
+            endpoint: The endpoint of the Azure AI Inference service deployment.
+            service_id: Service ID for the chat completion service. (Optional)
+            env_file_path: The path to the environment file. (Optional)
+            env_file_encoding: The encoding of the environment file. (Optional)
+            client: The Azure AI Inference client to use. (Optional)
+            model_info: The model info of the provided client. (Optional, required if client is provided)
+        """
+        if client is not None:
+            if model_info is None:
+                raise ServiceInitializationError("Model info is required when providing a client.")
+        else:
+            try:
+                azure_ai_inference_settings = AzureAIInferenceSettings.create(
+                    api_key=api_key,
+                    endpoint=endpoint,
+                    env_file_path=env_file_path,
+                    env_file_encoding=env_file_encoding,
+                )
+                client, model_info = self._create_client(azure_ai_inference_settings)
+            except ValidationError as e:
+                raise ServiceInitializationError(f"Failed to validate Azure AI Inference settings: {e}") from e
+
+        super().__init__(
+            ai_model_id=model_info.model_name,
+            service_id=service_id or model_info.model_name,
+            client=client,
+        )
+
+    async def generate_embeddings(self, texts: list[str], **kwargs: Any) -> ndarray:
+        """Generate embeddings from the Azure AI Inference service."""
+        settings: AzureAIInferenceEmbeddingPromptExecutionSettings = kwargs.get("settings", None)
+        response: EmbeddingsResult = await self.client.embed(
+            input=texts,
+            model_extras=settings.extra_parameters if settings else None,
+            dimensions=settings.dimensions if settings else None,
+            encoding_format=settings.encoding_format if settings else None,
+            input_type=settings.input_type if settings else None,
+            kwargs=kwargs,
+        )
+
+        return array([item.embedding for item in response.data])
+
+    def _create_client(
+        self, azure_ai_inference_settings: AzureAIInferenceSettings
+    ) -> tuple[EmbeddingsClientAsync, ModelInfo]:
+        """Create the Azure AI Inference client.
+
+        Client is created synchronously to check the model type before creating the async client.
+        """
+        embedding_client_sync = load_client_sync(
```

Comment:
Similar to Eduard's question: is there not an async client we can use?

Reply:
Please see the response above. We are returning an sync client eventually but we need to first check if the client is the right client.